Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -217,23 +226,99 @@ namespace MiscCmds
 		return true;
 	}
 
-	/** Throw the dice and tell all players within 6 km */
-	bool MiscCmds::UserCmd_Dice(uint iFromClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
-	{
-		wstring wscCharname = (const wchar_t*) Players.GetActiveCharacterName(iFromClientID);
-
-		int max = ToInt(GetParam(wscParam, ' ', 0));
-		if (max<=1)
-			max = 6;
-
-		uint number = (rand()%max)+1;
-		wstring wscMsg = set_wscDiceMsg;
-		wscMsg = ReplaceStr(wscMsg, L"%player", wscCharname);
-		wscMsg = ReplaceStr(wscMsg, L"%number", stows(itos(number)));
-		wscMsg = ReplaceStr(wscMsg, L"%max", stows(itos(max)));
-		PrintLocalUserCmdText(iFromClientID, wscMsg, set_iLocalChatRange);
-		return true;
-	}
+	/*
+	Roll dice for everyone within 6km of a vessel. Supports 1d20 formatting.
+	*/
+	bool MiscCmds::UserCmd_Dice(uint iFromClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+	{

+		boost::wregex expr(L"(\\d{1,2})[Dd](\\d{1,3})(([+\\-*])?(\\d{1,5}))?");
+		boost::wsmatch sm;
+
+		// If the regex finds a match denoting the correct roll format, run the randomized numbers
+		if (boost::regex_match(wscParam, sm, expr))
+		{
+
+			// Smatch index [1] represents the roll count
+			int rollCount = _wtoi(sm[1].str().c_str());
+
+			// Smatch index [2] represents the dice count
+			int diceCount = _wtoi(sm[2].str().c_str());
+
+			// Smatch index [3] represents any modifier numeric value. This is set ONLY if we are using a mod-operation
+			int modifierValue;
+
+			diceOperation operation;
+			if (sm[3].str().find(L"+") == 0)
+			{
+				operation = diceOperation::ADD;
+				modifierValue = _wtoi(sm[5].str().c_str());
+			}
+			else if (sm[3].str().find(L"-") == 0)
+			{
+				operation = diceOperation::SUBTRACT;
+				modifierValue = _wtoi(sm[5].str().c_str());
+			}
+			else
+			{
+				operation = diceOperation::NONE;
+			}
+
+			string diceResultSteps = "";
+			uint number = 0;
+
+			for (int i = 0; i < rollCount; i++)
+			{
+				int randValue = (rand() % diceCount) + 1;
+
+				// If we have a modifier, apply it
+				if (operation == diceOperation::ADD)
+				{
+					number += (randValue + modifierValue);

```

Comment:
You can just always add randValue to number as soon as you've got it, drop line 286, turn this into number += modifierValue and turn line 281 into number -= modifierValue

Reply:
I was aiming for consistency among all of the operations. Again. Readability. Instead of checking to see if the modifier value was positive or negative (thus requiring additional string parsing), I merely kept the modifier as a positive value. 

I suppose once I learned that the number was negative with the operation setter, the number could have been multiplied by -1, but the function overhead is quite non-existent. It didn't matter. Thus as a personal preference, it was set to what it is currently.

Reply:
ok